src/main/java/edu/cornell/kfs/coa/document/CuAccountGlobalMaintainableImpl.java

```diff
@@ -0,0 +1,186 @@
+        CuAccountGlobal accountGlobal = (CuAccountGlobal) getBusinessObject();
+        Map<String, Boolean> oldAccountClosedStatuses = getExistingAccountClosedStatuses(accountGlobal);
+        boolean closedStatusForLocks = getAccountClosedStatusForMaintenanceLocks(accountGlobal);
+        
+        Stream<MaintenanceLock> trickleDownLocks = accountGlobal.getAccountGlobalDetails().stream()
+                .map(accountDetail -> buildTemporaryAccountForMaintenanceLocks(accountDetail, closedStatusForLocks))
+                .filter(account -> isClosingAccount(account, oldAccountClosedStatuses))
+                .flatMap(this::generateTrickleDownMaintenanceLocks);
+        
+        return Stream.concat(accountLocks.stream(), trickleDownLocks)
+                .collect(Collectors.toUnmodifiableList());
+    }
+
+    private boolean getAccountClosedStatusForMaintenanceLocks(CuAccountGlobal accountGlobal) {
+        Boolean closed = accountGlobal.getClosed();
+        return closed != null && closed.booleanValue();
+    }
+
+    private Account buildTemporaryAccountForMaintenanceLocks(AccountGlobalDetail accountGlobalDetail, boolean closed) {
+        Account account = new Account();
+        account.setChartOfAccountsCode(accountGlobalDetail.getChartOfAccountsCode());
+        account.setAccountNumber(accountGlobalDetail.getAccountNumber());
+        account.setClosed(closed);
+        return account;
+    }
+
+    private Stream<MaintenanceLock> generateTrickleDownMaintenanceLocks(Account inactivatedAccount) {
+        List<MaintenanceLock> subAccountLocks = getSubAccountTrickleDownInactivationService()
+                .generateTrickleDownMaintenanceLocks(inactivatedAccount, getDocumentNumber());
+        List<MaintenanceLock> subObjectLocks = getSubObjectTrickleDownInactivationService()
+                .generateTrickleDownMaintenanceLocks(inactivatedAccount, getDocumentNumber());
+        List<MaintenanceLock> accountReversionLocks = getAccountReversionTrickleDownInactivationService()
+                .generateTrickleDownMaintenanceLocks(inactivatedAccount, getDocumentNumber());
+        return Stream.of(subAccountLocks, subObjectLocks, accountReversionLocks)
+                .flatMap(List::stream);
+    }
+
+    /*
+     * Copied and overrode this method to also inactivate downstream objects.
+     * Some of the updated logic is similar to that from the ObjectCodeGlobalMaintainableImpl class.
+     */
+    @Override
+    public void saveBusinessObject() {
+        BusinessObjectService boService = getBusinessObjectService();
+        GlobalBusinessObject gbo = (GlobalBusinessObject) getBusinessObject();
+
+        Map<String, Boolean> oldAccountClosedStatuses = getExistingAccountClosedStatuses((CuAccountGlobal) gbo);
+
+        // delete any indicated BOs
+        List<PersistableBusinessObject> bosToDeactivate = gbo.generateDeactivationsToPersist();
+        if (bosToDeactivate != null) {
+            if (!bosToDeactivate.isEmpty()) {
```

Comment:
Since the objects being checked are Lists, I would rather use CollectionUtils instead of ObjectUtils. I could simplify those spots to use one of the null-safe CollectionUtils methods if needed.